### DISCUSSION TITLE
clangd can not find include files

### DISCUSSION THREAD

**Discussion by akib35** [clangd/clangd#1517] [Q&A] [answered]
This extension was running fine for a long time. But today IDK why the clangd could not find my `bits/stdc++.h` file and started to show - 

> 'bits/stdc++.h' file not foundclang(pp_file_not_found)

Then I looked for the output logs, and there I found something this: 
```
[/home/[$username]/Documents/[$directory]]
/usr/lib/llvm-14/bin/clang -resource-dir=/usr/lib/llvm-14/lib/clang/14.0.0 -- /home/[$filename]
```
I found some idea from the response of this segment: 
https://github.com/clangd/clangd/issues/938#issuecomment-981173680

But adding these line in the `config.yaml` in `.config` did not solve the problem entirely. It still complains,

> In included file: 'cassert' file not foundclang(pp_file_not_found)
> stdc++.h(33, 10): Error occurred here

Then I added `-I//usr/include/c++/11/` in the configuration file where the header file may be found.
Finally, the problem was solved. Can someone tell me what happened actually?

**Comment by HighCommander4** [comment, marked as answer]
Are you running Ubuntu 22.04? If so, you may be running into #1394; see [this comment](https://github.com/clangd/clangd/issues/1394#issuecomment-1328676884) in particular for an explanation and workarounds.

**Reply by akib35** [reply]
i. i'm using Mint 21.1 Cinnamon
ii. the issue you mentioned, i've both gcc nd g++ preinstalled, and uninstalling gcc / g++ removes both package and vice versa
iii. i think there should be some functionality in the settings of clangd(directly for the extention , not using .config) to control include file paths

**Reply by HighCommander4** [reply]
Can you share [clangd logs](https://clangd.llvm.org/troubleshooting#gathering-logs)? That might help figure out what's going on in your case.

> i think there should be some functionality in the settings of clangd(directly for the extention , not using .config) to control include file paths

We try to keep most of clangd's functionality on the server side, to avoid duplicating code/logic between clients, since there is a different client for each editor.
